pkg/datapath/linux/node.go

```diff
@@ -744,10 +747,16 @@ func (n *linuxNodeHandler) insertNeighbor4(ctx context.Context, newNode *nodeTyp
 		IP:    nextHopIPv4,
 		IsNew: nextHopIsNew,
 	}
-	n.insertNeighborCommon(scopedLog, ctx, nh, link, refresh)
+
+	if err := errors.Join(errs, n.insertNeighborCommon(ctx, nh, link, refresh)); err != nil {
+		scopedLog.WithError(err).Debug("insert node neighbor IPv4 failed")
```

Comment:
Yes - previously this error was just logged hence figured this was not meant to be a hard error?

Reply:
I don't really see the value of debug level log because that means its visibility is only limited to users who've configured debug mode, which is not something recommended. Typically it's for developers. I think we can get rid of the log msg here in favor of the error.